pkg/crc/machine/generate_bundle_linux.go

```diff
@@ -0,0 +1,50 @@
+package machine
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/machine/bundle"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+func copyDiskImage(destDir string, crcBundleMetadata *bundle.CrcBundleInfo) error {
+	tmpDir := filepath.Join(constants.MachineCacheDir, "image-rebase")
+	_ = os.RemoveAll(tmpDir) // clean up before using it
+	if err := os.Mkdir(tmpDir, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		_ = os.RemoveAll(tmpDir) // clean up after using it
+	}()
+
+	// Copy the disk image from the machine/crc folder
+	diskName := filepath.Base(crcBundleMetadata.GetDiskImagePath())
+	diskPath := filepath.Join(constants.MachineInstanceDir, constants.DefaultName, diskName)
```

Comment:
probably better if we can avoid hardcoding `DefaultName` here? I suspect this is poking at internals of the machine driver? Or are there other places of `crc` code which know about `~/.crc/machines/crc/crc.qcow2`? At the very least, I'd move this `filepath.Join` in a `GetVMDiskImagePath()` in one of the `machine` go modules. But maybe more can be done, I did not open this code during the review?

Reply:
Agree, but as of now this `DefaultName` is used lot of places and might be a refactor needed if we want to avoid that completely or you are just thinking about this specific part of code?

Reply:
Thinking of this PR, and this hunk in particular since this is where I noticed it.